File: mobileBE/src/main/java/com/uth/mobileBE/dto/response/LoanDetailResponse.java
package com.uth.mobileBE.dto.response;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.uth.mobileBE.models.enums.StatusLoanDetail;
import lombok.Builder;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@Builder
public class LoanDetailResponse {

    // BỔ SUNG: ID của chính chi tiết mượn này (Rất quan trọng để gọi API trả sách)
    private Long loanDetailId;

    private Long loanId;
    private Long copyId;

    // Thông tin kèm theo để hiển thị lên UI cho đẹp mà không cần gọi API khác
    private Long bookId;
    private String bookTitle;

    private String author;
    private String category;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime dueDate;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime returnDate;

    private StatusLoanDetail status;
    private Double penaltyAmount;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createdAt;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime updateAt;
}

File: app/src/main/java/com/example/quanlythuvien/data/model/response/LoanDetailResponse.kt
package com.example.quanlythuvien.data.model.response

import com.google.gson.annotations.SerializedName

data class LoanDetailResponse(
    @SerializedName("loanDetailId") val loanDetailId: Long, // QUAN TRỌNG: ID mới để gọi API
    @SerializedName("loanId") val loanId: Long,
    @SerializedName("copyId") val copyId: Long,
    @SerializedName("bookId") val bookId: Long?,
    @SerializedName("bookTitle") val bookTitle: String?,
    @SerializedName("author") val author: String?,
    @SerializedName("category") val category: String?,
    @SerializedName("dueDate") val dueDate: String?,
    @SerializedName("returnDate") val returnDate: String?,
    @SerializedName("status") val status: String,
    @SerializedName("penaltyAmount") val penaltyAmount: Double?,
    @SerializedName("createdAt") val createdAt: String?,
    @SerializedName("updateAt") val updateAt: String?
)